app/api/v2/commercial/accounts.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+module API
+  module V2
+    module Commercial
+      # Admin functionality over abilities
+      class Accounts < Grape::API
+        resource :accounts do
+          helpers ::API::V2::NamedParams
+          helpers do
+            def permitted_search_params(params)
+              params.slice(:keyword)
+            end
+          end
+
+          desc 'Returns array of accounts which user have access as paginated collection',
+               failure: [
+                 { code: 404, message: 'Account not found' }
+               ],
+               success: API::V2::Commercial::Entities::Account
+          params do
+            use :pagination_filters
+          end
+
+          get do
+            # Check account in the organization that user belong to
+            members = Membership.joins('LEFT JOIN organizations ON organizations.id = memberships.organization_id')
+                                .where(user_id: current_user.id)
+                                .select('memberships.*,organizations.name, organizations.organization_id')
+                                .pluck(:organization_id, :'organizations.name', :'organizations.organization_id')
+                                .map { |id, name, pid| { id: id, name: name, pid: pid } }
+            error!({ errors: ['identity.member.not_found'] }, 404) if members.nil? || members.length.zero?
+
+            # Check user is barong organization admin or not
+            if members.first[:id].zero?
```

Comment:
this is very suspicious to me, what is the logic?

Reply:
I set the admin who can access to all organization with `organization_id: 0`. So I check the level from here.

Reply:
OK. I will change the logic of check admin of all organization to be check from user role instead.